samples/hello/README.rst

```diff
@@ -19,29 +19,73 @@ Building and Running
 Authentication specific configuration
 =====================================
 
-PSK based auth
---------------
+Golioth offers two `authentication methods`_: Pre-Shared Keys (PSK) or Public
+Key Cryptography using Certificates (certs). Normally, it is the responsibility
+of the Golioth SDK user to load these credentials at runtime. For simplicity,
+we provide facilities to hardcode these credentials or set them at
+runtime for our samples.
+
+PSK based auth - Hardcoded
+--------------------------
 
 Configure the following Kconfig options based on your Golioth credentials:
 
-- GOLIOTH_SYSTEM_CLIENT_PSK_ID  - PSK ID of registered device
-- GOLIOTH_SYSTEM_CLIENT_PSK     - PSK of registered device
+- ``GOLIOTH_SAMPLE_HARDCODED_PSK_ID``  - PSK ID of registered device
+- ``GOLIOTH_SAMPLE_HARDCODED_PSK``     - PSK of registered device
```

Comment:
Please do this kind of improvement in separate commit, so it is less overwhelming to review logical changes.